src/Storages/StorageView.cpp

```diff
@@ -33,8 +33,10 @@ namespace ErrorCodes
 StorageView::StorageView(
     const StorageID & table_id_,
     const ASTCreateQuery & query,
-    const ColumnsDescription & columns_)
+    const ColumnsDescription & columns_,
+    const Settings & settings)
     : IStorage(table_id_)
+    , settings_changes{{"join_use_nulls", Field(settings.join_use_nulls)}}
```

Comment:
Maybe better to save all changed settings, not only join_use_nulls?

Reply:
Changed to `settings.changes()`, but I'm not sure that it won't have any unexpectable side-effects, can we imagine some setting shouldn't be saved?

Reply:
> unexpectable side-effects

`join_use_nulls` case is special here because it can break in unclear ways.

Other settings should probalby not be saved, unless there are known similar problems.

And I'm not sure whether it should be possible to do `select ... settings http_send_timeout = ...`, especially in a subquery or view.